crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -225,6 +226,48 @@ fn assert_component_access_compatibility(
     panic!("error[B0001]: Query<{query_type}, {filter_type}> in system {system_name} accesses component(s) {accesses} in a way that conflicts with a previous system parameter. Consider using `Without<T>` to create disjoint Queries or merging conflicting Queries into a `ParamSet`.");
 }
 
+unsafe impl<Q: QueryTermGroup + 'static, F: QueryTermGroup + 'static> SystemParam
+    for TermQuery<'_, '_, Q, F>
+{
+    type State = TermQueryState<Q, F>;
+    type Item<'w, 's> = TermQuery<'w, 's, Q, F>;
+
+    fn init_state(world: &mut World, system_meta: &mut crate::system::SystemMeta) -> Self::State {
+        let state = TermQueryState::new(world);
+        assert_component_access_compatibility(
+            &system_meta.name,
+            "term_query",
+            "term_query",
```

Comment:
```suggestion
            "term_query_filter",
```

I think this is both easier to understand, and will produce more helpful error messages.